lib/src/client/client.dart

```diff
@@ -17,19 +17,60 @@ import 'dart:async';
 
 import 'call.dart';
 import 'channel.dart';
+import 'common.dart';
+import 'interceptor.dart';
 import 'method.dart';
 
 /// Base class for client stubs.
 class Client {
   final ClientChannel _channel;
   final CallOptions _options;
+  final List<ClientUnaryInterceptor> _unaryInterceptors;
+  final List<ClientStreamingInterceptor> _streamingInterceptors;
 
-  Client(this._channel, {CallOptions options})
-      : _options = options ?? CallOptions();
+  Client(this._channel,
+      {CallOptions options,
+      Iterable<ClientUnaryInterceptor> unaryInterceptors,
+      Iterable<ClientStreamingInterceptor> streamingInterceptors})
+      : _options = options ?? CallOptions(),
+        _unaryInterceptors =
+            List.unmodifiable(unaryInterceptors ?? Iterable.empty()),
```

Comment:
Question for my own understanding:

1. Why do we create a new List object instead of simply using the Iterable passed in?

2. Should we allow the user to determine what collection to pass in, instead of forcing List type?

Reply:
1. If you take interables and keep them for subsequent use it is a good idea to copy them into lists to prevent some strange modifications to the iterables from affecting your code. (This is also not-likely to be a performance sensitive code so copying is fine).
2. Users can pass anything right now (because we choose `Iterable` interface as a parameter). This is a bit incosistent with other cases in the gRPC code base where we force users to pass `List` (e.g. server interceptors). I don't have any strong opinion on that though.

Reply:
Yeah I figured if we did not need to make a copy, we could just pass the Iterable as is, instead of forcing List type. But given point 1, point 2 thus makes sense.

Reply:
Can this stay as it is now then?